controllers/configmap_controller.go

```diff
@@ -512,9 +512,37 @@ func (r *ConfigMapReconciler) reconcileProxyCerts(ctx context.Context, trustedCA
 	if err := r.ensureProxyCertsCMIsValid(ctx, trustedCA.GetLabels()[InjectionRequestLabel]); err != nil {
 		return err
 	}
+	return r.ensureTrustedCABundleInNodes(ctx, trustedCA.Data)
+}
+
+// ensureTrustedCABundleInNodes copies over the trust CA bundle onto each Windows instance in the cluster
+func (r *ConfigMapReconciler) ensureTrustedCABundleInNodes(ctx context.Context, caData map[string]string) error {
+	winNodes, err := r.k8sclientset.CoreV1().Nodes().List(ctx, meta.ListOptions{LabelSelector: nodeconfig.WindowsOSLabel})
+	if err != nil {
+		return fmt.Errorf("error listing nodes: %w", err)
+	}
+	for _, node := range winNodes.Items {
+		if err := r.ensureTrustedCABundleInNode(ctx, caData, node); err != nil {
+			return err
+		}
+	}
 	return nil
 }
 
+// ensureTrustedCABundleInNodes places the trusted CA bundle data into a file on the given node
+func (r *ConfigMapReconciler) ensureTrustedCABundleInNode(ctx context.Context, caData map[string]string, node core.Node) error {
+	winInstance, err := r.instanceFromNode(&node)
+	if err != nil {
+		return fmt.Errorf("error creating instance for node %s: %w", node.Name, err)
+	}
+	nc, err := nodeconfig.NewNodeConfig(r.client, r.k8sclientset, r.clusterServiceCIDR, r.watchNamespace,
+		winInstance, r.signer, nil, nil, r.platform)
+	if err != nil {
+		return fmt.Errorf("error creating nodeConfig for instance %s: %w", winInstance.Address, err)
+	}
+	return nc.UpdateTrustedCABundleFile(caData)
```

Comment:
The error messages here don't need to each include the node/instance details. that can be wrapped at the call site at the first place in the chain which is dealing with this node object, in this case within`ensureTrustedCABundleInNodes`

This also helps to prevent stuttering, giving a cleaner error message.